### DISCUSSION TITLE
Multilingual problem

### DISCUSSION THREAD

**Discussion by ling-zhan** [guoyingtao/Mantis#123] [Q&A]
Hi, How to use the resources language file to change the text of the "Down" or "Cancel" and "Reset" buttons.

PS: I use the SwiftUI language, using UIViewControllerRepresentable return a Mantis.CropViewController.

**Comment by guoyingtao** [comment]
It seems the localization is broken for swiftUI example. I will fix it soon.

**Reply by guoyingtao** [reply]
@ling-zhan I found a solution from this [link](https://forums.swift.org/t/swift-package-manager-localization/46685/5)
![image](https://user-images.githubusercontent.com/26723384/128088024-610a8ebc-4b06-4340-98f7-448f5100ea24.png)

**Reply by ling-zhan** [reply]
Thank you for your reply and look forward to your repair.
I will try to see if this method works.

**Reply by guoyingtao** [reply]
@ling-zhan The "fix" I mean is to try the solution I mentioned above. I tried it in the SwiftUI example and it worked.

**Reply by ling-zhan** [reply]
Got it, thank you. ^^"

**Comment by hajducak** [comment]
Hey guys, nice fix. can you guys get any example here, or apply it to example cases, because I tried and for me it is not working. maybe I am doing something wrong. Thanks

**Comment by bleckett** [comment]
Very nice tool - thank you for making.
One suggestion concerning transparent images...
I'm changing the view controller's background to white (if there are dark pixels) when detecting transparency, but I was thinking it would be nice to have a layer underneath the image that is white or the famous checkerboard that has the same dimensions of the image when cropping?

This is what I use for creating a grid

`
func createGridImage(size: CGSize, rows: Int) -> UIImage {
    let renderer = UIGraphicsImageRenderer(size: size)
    let img = renderer.image { ctx in
        let squareSize = size.width / CGFloat(rows)
        
        for row in 0..<rows {
            for col in 0..<rows {
                if (row + col) % 2 == 0 {
                    ctx.cgContext.setFillColor(UIColor.white.cgColor)
                } else {
                    ctx.cgContext.setFillColor(kCOLOR.GRID_GREY.cgColor)
                }
                let rect = CGRect(x: CGFloat(col) * squareSize, y: CGFloat(row) * squareSize, width: squareSize, height: squareSize)
                ctx.cgContext.addRect(rect)
                ctx.cgContext.drawPath(using: .fill)
            }
        }
    }
    return img
}
`

And for transparency I am using this...

`
extension UIImage {
        
    func hasTransparency() -> Bool {
        guard let cgImage = self.cgImage else { return false }
        let alphaInfo = cgImage.alphaInfo

        // No alpha channel means no transparency
        if alphaInfo == .none || alphaInfo == .noneSkipFirst || alphaInfo == .noneSkipLast {
            return false
        }

        let scaleFactor = 0.1 // Downscale by 90%
        let scaledWidth = Int(CGFloat(cgImage.width) * scaleFactor)
        let scaledHeight = Int(CGFloat(cgImage.height) * scaleFactor)
        let bytesPerPixel = 4
        let bytesPerRow = bytesPerPixel * scaledWidth
        let bitsPerComponent = 8

        var pixelData = [UInt8](repeating: 0, count: scaledWidth * scaledHeight * bytesPerPixel)

        guard let context = CGContext(data: &pixelData,
                                      width: scaledWidth,
                                      height: scaledHeight,
                                      bitsPerComponent: bitsPerComponent,
                                      bytesPerRow: bytesPerRow,
                                      space: CGColorSpaceCreateDeviceRGB(),
                                      bitmapInfo: CGImageAlphaInfo.premultipliedLast.rawValue) else {
            return false
        }

        context.draw(cgImage, in: CGRect(x: 0, y: 0, width: scaledWidth, height: scaledHeight))

        for i in stride(from: 0, to: pixelData.count, by: bytesPerPixel) {
            let alpha = pixelData[i + 3]
            if alpha < 255 {
                return true
            }
        }

        return false
    }

}
`

I was thinking of doing a fork, but I've never really worked in Github before....

**Reply by bleckett** [reply]
hmmm sorry - the code blocks aren't working very well for me...
